### PR TITLE
Run HttpUrl tests on all platforms

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: okhttp/src/jvmTest/java/okhttp3/HttpUrlJvmTest.kt [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+import java.net.URI
+import java.net.URL
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.HttpUrl.Companion.toHttpUrlOrNull
+import okhttp3.testing.PlatformRule
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+
+@Suppress("HttpUrlsUsage") // Don't warn if we should be using https://.
+open class HttpUrlJvmTest {
```
Comment (swankjesse) [inline okhttp/src/jvmTest/java/okhttp3/HttpUrlJvmTest.kt:27]: Nothing new here; just split the JVM-specific stuff off of the HttpUrlTest

File: okhttp/src/jvmTest/java/okhttp3/UrlComponentEncodingTesterJvm.kt [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+ * limitations under the License.
+ */
+package okhttp3
+
+import assertk.fail
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.HttpUrl.Companion.toHttpUrlOrNull
+import okhttp3.UrlComponentEncodingTester.Component
+
+actual fun urlComponentEncodingTesterJvmPlatform(
```
Comment (swankjesse) [inline okhttp/src/jvmTest/java/okhttp3/UrlComponentEncodingTesterJvm.kt:23]: I extracted this out from the corresponding tests!

File: okhttp-testing-support/src/commonMain/kotlin/okhttp3/TestUtilCommon.kt [hunk 3]
```diff
@@ -27,3 +28,11 @@ expect val SYSTEM_FILE_SYSTEM: FileSystem
 expect fun getEnv(name: String): String?
 
 expect val isJvm: Boolean
+
+fun String(vararg codePoints: Int): String {
```
Comment (yschimke) [inline okhttp-testing-support/src/commonMain/kotlin/okhttp3/TestUtilCommon.kt:32]: I can't believe it's come to this...

**Review by yschimke — APPROVED** [review]

File: okhttp-testing-support/src/commonMain/kotlin/okhttp3/TestUtilCommon.kt (hunk 3 above)
Comment (swankjesse) [reply okhttp-testing-support/src/commonMain/kotlin/okhttp3/TestUtilCommon.kt:32]: If you wish to make an apple pie from scratch, you must first invent the universe.

Comment (yschimke) [reply okhttp-testing-support/src/commonMain/kotlin/okhttp3/TestUtilCommon.kt:32]: 🍎 + 🥧 = �

**Pull request merged**